plugins/modules/ag_listener.ps1

```diff
@@ -15,7 +15,7 @@ $spec = @{
     options = @{
         ag_name = @{type = 'str'; required = $true }
         listener_name = @{type = 'str'; required = $true }
-        ip_address = @{type = 'str'; required = $false }
+        ip_address = @{type = 'list'; required = $false }
         subnet_ip = @{type = 'str'; required = $false }
         subnet_mask = @{type = 'str'; required = $false; default = '255.255.255.0' }
```

Comment:
> > I think you will need to update the type of the subnet_mask too.
> 
> I think you are right that the change will be needed there as well. I do not have a use case to test that change and left it out of my request due to that

Yeah, all three of these to be lists to be consistent with the [inputs in DBATools](https://github.com/dataplat/dbatools/blob/6c7cb4bb1a7377a4485df8f96d74ab7e642c32b8/public/Add-DbaAgListener.ps1#L88-L90):

* IP Address
* Subnet IP
* Subnet Mask